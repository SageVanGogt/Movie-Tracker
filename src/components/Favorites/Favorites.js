import React, { Component } from 'react';
import {connect} from 'react-redux';
import Card from '../Card/Card';
import { getUserFavorites } from '../../apiCall/apiCall';

class Favorites extends Component {
  constructor(props) {
    super(props);
    this.state = {
      favorites: []
    }

  }

componentDidMount() {
 this.populateFavorites();
}

populateFavorites = async () => {
  debugger 
  const favorites = await getUserFavorites(this.props.user.user_id);
  this.setState({
    favorites
  })
}

render(){
  return(
    <div>favorites</div>
  )
} 
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps, null)(Favorites);