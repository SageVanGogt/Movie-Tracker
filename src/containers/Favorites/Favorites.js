import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../../components/Card/Card';
import { getUserFavorites } from '../../apiCall/apiCall';
import { addFavoritesToStore } from './../../actions/index'

class Favorites extends Component {
  constructor(props) {
    super(props);
    // this.state = {
    //   favorites: []
    // }
  }

  componentDidMount() {
    this.getFavorites();
  }

  getFavorites = async () => {
    const favorites = await getUserFavorites(this.props.user.user_id);
    this.props.populateFavorites(favorites.data)
  }

  render() {
    return (
      <div>favorites</div>
    )
  } 
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  populateFavorites: (movies) => dispatch(addFavoritesToStore(movies))
})
export default connect(mapStateToProps, mapDispatchToProps)(Favorites);