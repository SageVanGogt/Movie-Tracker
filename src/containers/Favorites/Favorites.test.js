import React from 'react';
import { Favorites, mapStateToProps } from './Favorites';
import { shallow } from 'enzyme';

describe('Favorites', () => {
  describe('mapStateToProps', () => {
    it('should map user to props', () => {
      const mockState = {
        user: {   
          user_id: 1,
          name: 'Sup Dude'
        },
        films: [],
        favorites: [{},{}]
      }
      const expected = {
        user: {
        user_id: 1,
        name: 'Sup Dude'
        }, 
        favorites: [{},{}]
      }
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    })
  })
})