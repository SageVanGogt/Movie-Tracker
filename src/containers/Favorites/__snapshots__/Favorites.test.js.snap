// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    favorites={
        Array [
            Object {
              "movie_id": 1,
            },
            Object {
              "movie_id": 2,
            },
            Object {
              "movie_id": 3,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component)
          movie_id={1}
/>,
        <Connect(Component)
          movie_id={2}
/>,
        <Connect(Component)
          movie_id={3}
/>,
      ],
      "className": "favorites-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "movie_id": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "movie_id": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "movie_id": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component)
            movie_id={1}
/>,
          <Connect(Component)
            movie_id={2}
/>,
          <Connect(Component)
            movie_id={3}
/>,
        ],
        "className": "favorites-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "movie_id": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "movie_id": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "movie_id": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
